### DISCUSSION TITLE
ent contrib template naming

### DISCUSSION THREAD

**Discussion by cliedeman** [ent/ent#1464] [Ideas]
Should the templates ont he contrib libraries be prefixed with their package? 

For example
* collection -> entgqlcollection
* edge -> entgqledge

etc.

This will help to prevent future template name collisions

**Comment by a8m** [comment]
Agree with this, and actually already talk about it with @rotemtam.

I plan to add an extension/plugin option to entc (as describe [here](https://github.com/ent/ent/pull/1441#discussion_r609976324)) and then we'll add this change using a [codegen hook](https://entgo.io/docs/code-gen#code-generation-hooks), to make migration smoother.
